sktime/datatypes/_series/_check.py

```diff
@@ -74,6 +75,23 @@ def check_pddataframe_series(obj, return_metadata=False, var_name="obj"):
     if _req("feature_names", return_metadata):
         metadata["feature_names"] = obj.columns.to_list()
 
+    if _req("feature_kind", return_metadata):
+        col_dtypes = obj.dtypes.to_list()
+        for i, dtype in enumerate(col_dtypes):
+            if dtype == float:
```

Comment:
Yes, exactly this is the kind of logic needed. Since we will repeat this, to ensure we are DRY ("don't repeat yourself") I would suggest:

* deal with this by a `dict` instead of `if` / `elif` chain
* move the full logic private utility function in the `_dtypekind` module, import from there. E.g., `pandas_dtype_to_kind`
* calls to this should be one-liners, applicable to objects with a `column` variable

I was also wondering whether `pandas` has any utility method like this as it supports the interchange protocol, but I was able to find only "from" logic, not "to" logic. Maybe you have more luck if you decide to search a bit.

Reply:
I've moved the function to _dtypekind.py.
But, I couldn't do away with the if/elif statements. A dict does not work due to how the keys are being compared. Here is an example:
```python
s = pd.Series([1, 4, 0.5, -3], dtype=np.float64 ,name="a")
print(s.dtype==float) #this evaluates to true so can be used in the if/elif

dtype_to_kind = {
    float : DtypeKind.FLOAT,
    int : DtypeKind.INT,
    np.uint : DtypeKind.UINT,
    object : DtypeKind.CATEGORICAL,
    bool : DtypeKind.BOOL,
    pd.DatetimeIndex : DtypeKind.DATETIME
}

dtype_to_kind[s.dtype] # key error here
```

If we still want to use a dict then it would have to contain all kinds of dtypes as keys like `np.float16`,`np.float32`,`np.uint8`, etc....

Reply:
I do not follow - what exactly is the problem?

Reply:
So, it is okay the way it is now right?

Reply:
I would say, no, because it does not cover the possible `dtype`-s exhaustively. The `pandas.api` functions might be better suited here.

Reply:
Yes, they seem to cover a wider set of dtypes. Will do.